### PR TITLE
Feature/set component position method

### LINKED ISSUE #24: Organize component without position
## Description

**As a** user,

**I want** all my components are organize

**so that** see correctly all of them.

## Acceptance Criteria

 **Scenario** | First draw of the components need to be organize
 ------------ | -------------------------------------------------
 **Given**    | I parse for the first time a folder
 **When**     | I draw all components
 **Then**     | I expect all components are organize

## Notes

We want to have a grid of component with the width of the grid is automaticelly calculated with the current size of the canvas.

The width and height of the cell is calculated with the maxinum size of all componnent.

It's the first version of the placement algorithm.

### PR DESCRIPTION
**Pull request by Hundraw**
:information_source:  Don't forget to modify the changelog.md before merging this branch.

Close #24

### CODE REVIEW

File: src/draw/DefaultDrawer.js [hunk 1]
```diff
@@ -48,25 +53,96 @@ class DefaultDrawer {
...
+  getComponentSize(svgItem) {
+    const bBox = this.d3.select(svgItem).node().getBBox();
+    return { width: bBox.width, height: bBox.height };
+  }
+
+  /**
+  * Draw component with default template.
+  * @param {Object} svgList - HTML elements selected by D3.js.
+  */
+  drawDefaultResource(svgList) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:140]: svgList is not adapted, please use `componentElement` or else

File: tests/unit/draw/DefaultDrawer.spec.js [hunk 2]
```diff
@@ -31,40 +32,164 @@ describe('Test Class: DefaultDrawer()', () => {
...
 
-  describe('Test method: drawDefaultResource', () => {
+  describe('Test method: initializeComponents', () => {
+    let drawer;
+    let component01;
+    let component02;
+    let svgList;
+
+    beforeEach(() => {
+      document.body.innerHTML = '<svg id="viewport"></svg>';
```
Comment (Zorin95670) [inline tests/unit/draw/DefaultDrawer.spec.js:54]: not sure it's needed

File: changelog.md [hunk 3]
```diff
@@ -16,8 +16,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improve all class models and their individual tests.
 - Remove function renderString for SVG template.
 - Create default class structure draw/metadata/parser/render and their individual tests.
-- Add default error for parser
-- Add `Draw` and `DrawDefaultContent` methods to DefaultDrawer.js
+- Add default error for parser.
+- Add `Draw` and `DrawDefaultContent` methods to DefaultDrawer.js.
```
Comment (Zorin95670) [inline changelog.md:20]: Bad method name

File: changelog.md [hunk 4]
```diff
@@ -16,8 +16,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Add `getComponentSize`, `setComponentSize`, `setComponentPosition` and `initializeComponent`
```
Comment (Zorin95670) [inline changelog.md:21]: Prefere to say, add functions to set default component position

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: tests/unit/draw/DefaultDrawer.spec.js (hunk 2 above)
Comment (Hundraw) [reply tests/unit/draw/DefaultDrawer.spec.js:54]: We need it because the tests call for it and it can't be empty.

File: src/draw/DefaultDrawer.js [hunk 5]
```diff
@@ -31,11 +37,10 @@ class DefaultDrawer {
     const componentsSelection = this.d3.select(parentId)
       .selectAll('.component')
       .data(components, (data) => data.id);
-    const component = componentsSelection.enter()
-      .append('svg');
+    const svgList = componentsSelection.enter().append('svg');
 
-    component.attr('class', 'component');
-    component.each((_data, index, array) => {
+    svgList.attr('class', 'component');
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/draw/DefaultDrawer.js]: you forget to renamme svgList in test :)

File: src/draw/DefaultDrawer.js [hunk 6]
```diff
@@ -48,25 +53,97 @@ class DefaultDrawer {
...
+    component.drawOption.x = indexX * (width + this.margin);
+    component.drawOption.y = indexY * (height + this.margin);
+  }
+
+  /**
+   * Set size of the component from the reel size of the svg model.
+   * @param {Object} svgIndex - HTML elements selected by D3.js.
+   * @param {Component} component - Component object we want to set the size.
+   */
+  storeComponentSize(svgIndex, component) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:121]: Rename svgIndex to svgItem

File: src/draw/DefaultDrawer.js [hunk 7]
```diff
@@ -48,25 +53,97 @@ class DefaultDrawer {
...
+        component,
+        index,
+        viewportWidth,
+        componentMaxSize,
+      ));
+    }
+  }
+
+  /**
+   * Defines the position of the components without drawOption by computing
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:100]: remove `without drawOption`, because in future we can call these function to reset all components position :)

**Review by Zorin95670 — APPROVED** [review]
